sru/modules.py

```diff
@@ -3,230 +3,78 @@
-            input sequence element. For example, if the input sequence
-            is a sequence of word embeddings, `input_size` is the
-            dimensionality of a single word embedding, e.g. 300.
-        hidden_size (int) : the dimensionality of the hidden state
-            of this cell.
-        dropout (float) : a number between 0.0 and 1.0. The amount of dropout
-            applied to `g(c_t)` internally in this cell.
-        rnn_dropout (float) : the amount of dropout applied to the input of
-            this cell.
-        use_tanh (bool) : use tanh activation
-        is_input_normalized (bool) : whether the input is normalized (e.g. batch norm / layer norm)
-        bidirectional (bool) : whether or not to employ a bidirectional cell.
+    A single SRU layer as per `LSTMCell`, `GRUCell` in Pytorch.
     """
 
+    __constants__ = ['input_size', 'hidden_size', 'output_size', 'rnn_dropout',
+                     'dropout', 'bidirectional', 'has_skip_term', 'highway_bias',
+                     'v1', 'rescale', 'activation_type', 'activation', 'custom_m',
+                     'projection_size', 'num_matrices', 'layer_norm', ]
+
     def __init__(self,
-                 input_size,
-                 hidden_size,
-                 dropout=0,
-                 rnn_dropout=0,
-                 bidirectional=False,
-                 n_proj=0,
-                 use_tanh=0,
-                 highway_bias=0,
-                 has_skip_term=True,
-                 layer_norm=False,
-                 rescale=True,
-                 v1=False,
-                 custom_m=None):
+                 input_size: int,
+                 hidden_size: int,
+                 dropout: float = 0.0,
+                 rnn_dropout: float = 0.0,
+                 bidirectional: bool = False,
+                 n_proj: int = 0,
+                 use_tanh: bool = False,
+                 highway_bias: float = 0.0,
+                 has_skip_term: bool = True,
+                 layer_norm: bool = False,
+                 rescale: bool = True,
+                 v1: bool = False,
+                 custom_m: Optional[nn.Module] = None):
+        """Initialize the SRUCell module.
+
+        Parameters
+        ----------
+        input_size: int
+            the number of features in the input `x`
+        hidden_size: int
+            the number of features in the hidden state *for each direction*
+        dropout: float, optional
+            the dropout value applied between layers (default=0)
+        rnn_dropout: float, optional
+            [DEPRECATED] the variational dropout value (default=0)
+        bidirectional: bool, optional
+            if True, set the module as a bidirectional SRU (default=False)
+        n_proj: int, optional
+            if non-zero, factorize the ``weight`` parameter matrix as a product of two
+            parameter matrices, using an innder dimension ``n_proj`` (default=0)
+        use_tanh: bool, optional
```

Comment:
for the deprecated params (rnn_dropout, use_tanh, v1) is it worth adding a note around the "correct" way to achieve the same behavior? E.g. if I wanted to use tanh activation, is there a different flag I should set instead?

Reply:
Also does deprecated imply these params are not torchscript compatible?

Reply:
I tend to agree that it might be useful to have a brief description of why the functionality is being removed altogether, or how to achieve the functionality using a different approach.

Reply:
@jshapiro-asapp @hpasapp  I marked them as deprecated because i don't want users to use the options anymore:
(1) the options are for backward compatibility since v1 and won't give better performance;
(2) i want to remove these options and therefore i can clean up the cpp/cu code.

in other words, the "correct" way is not to change the default value and not use these options anymore.
any thoughts?

Reply:
"tanh is deprecated because minimal performance improvement, and increases codebase size. This option will be removed at the next major version upgrade.
"rnn_dropout is deprecated because minimal performance improvement, and increases codebase size. This option will be removed at the next major version upgrade"